google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/HeaderInterceptor.java

```diff
@@ -197,4 +205,13 @@ private Attributes getMetricAttributes(String key, String method, DatabaseName d
           return attributesBuilder.build();
         });
   }
+
+  private void addBuiltInMetricAttributes(
+      CompositeTracer compositeTracer, DatabaseName databaseName) {
+    // Built in metrics Attributes.
+    compositeTracer.addAttributes(
```

Comment:
Strictly speaking, we shouldn't add these attributes to the `compositeTracer`, but only to the underlying built-in metrics tracer. Now the attributes will also be added to any other `MetricsTracer` that is part of the composite tracer, but it only happens if you have built-in tracing enabled.
Meaning that:
1. Enabling built-in metrics will obviously enable built-in metrics.
2. But it will also have the side-effect that any other metrics-tracer that you had enabled will suddenly get two additional attributes.

Is there any way that we can avoid that side-effect?

Reply:
I didn't quite get you. We have a check in the `CompositeTracer` to pass these attributes further only for `MetricsTracer`.  We only have one MetricsTracer

```
public void addAttributes(String key, String value) {
    for (ApiTracer child : children) {
      if (child instanceof MetricsTracer) {
        MetricsTracer metricsTracer = (MetricsTracer) child;
        metricsTracer.addAttributes(key, value);
      }
    }
  };
  ```

Reply:
Cannot a user create and add a `MetricsTracer` themselves?

Reply:
This class is decorated with @InternalApi annotation . So we don't recommend customers to directly use this class. But even if they use this class to register metricstracerfactory, then yes they would get extra attributes.